core/tests/aot/fx_programs_test.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2024 Advanced Micro Devices, Inc
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from pathlib import Path
+import tempfile
+
+import pytest
+import torch
+
+from shark_turbine.aot import (
+    FxPrograms,
+    FxProgramsBuilder,
+)
+
+
+def test_save_load_dynamic_shapes():
+    if torch.__version__ < "2.3.0.dev1":
+        pytest.skip("Unsupported PyTorch version")
+
+    class M(torch.nn.Module):
+        def __init__(self):
+            super().__init__()
+
+            self.branch1 = torch.nn.Sequential(torch.nn.Linear(64, 32), torch.nn.ReLU())
+            self.branch2 = torch.nn.Sequential(
+                torch.nn.Linear(128, 64), torch.nn.ReLU()
+            )
+            self.buffer = torch.ones(32)
+
+        def forward(self, x1, x2):
+            out1 = self.branch1(x1)
+            out2 = self.branch2(x2)
+            return (out1 + self.buffer, out2)
+
+    example_args = (torch.randn(32, 64), torch.randn(32, 128))
+
+    # Create a dynamic batch size
+    batch = torch.export.Dim("batch")
+    # Specify that the first dimension of each input is that batch size
+    dynamic_shapes = {"x1": {0: batch}, "x2": {0: batch}}
+
+    fxb = FxProgramsBuilder(M())
+
+    @fxb.export_program(args=example_args, dynamic_shapes=dynamic_shapes)
+    def dynamic_batch(module: M, x1, x2):
+        return module.forward(x1, x2)
+
+    @fxb.export_program(args=example_args)
+    def bs1(module: M, x1, x2):
```

Comment:
Does the dynamic shapes annotation here replace the 32s from example_args or does it add a dimension? if the former, 
should `def bs1` be `def bs32` for clarity?